name: Backport Issue Check

on:
  pull_request_target:
    types:
      - edited
      - opened
      - reopened
      - synchronize
    branches:
      - v*-branch

permissions:
  contents: read

jobs:
  backport:
    name: Backport Issue Check
    concurrency:
      group: backport-issue-check-${{ github.ref }}
      cancel-in-progress: true
    runs-on: ubuntu-24.04
    if: github.repository == 'zephyrproject-rtos/zephyr'
    permissions:
      issues: read # to check if associated issue exists for backport

    steps:
      - name: Check out source code
        uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0

      - name: Set up Python
        uses: actions/setup-python@e797f83bcb11b83ae66e0230d6156d7c80228e7c # v6.0.0
        with:
          python-version: 3.12

      - name: Install uv
        uses: astral-sh/setup-uv@b75a909f75acd358c2196fb9a5f1299a9a8868a4 # 6.7.0
        with:
          enable-cache: true
          cache-dependency-glob: "pyproject.toml"

      - name: Install Python packages
        run: |
          uv sync --group actions

      - name: Run backport issue checker
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        run: |
          uv run ./scripts/release/list_backports.py \
          -o ${{ github.event.repository.owner.login }} \
          -r ${{ github.event.repository.name }} \
          -b ${{ github.event.pull_request.base.ref }} \
          -p ${{ github.event.pull_request.number }}
